src/Utils/AST.php

```diff
@@ -230,9 +259,10 @@ static function astFromValue($value, InputType $type)
                 return new IntValueNode(['value' => $serialized]);
             }
             if (is_float($serialized)) {
-                if ((int) $serialized == $serialized) {
+                if ($serialized <= PHP_INT_MAX && floor($serialized) === $serialized) {
```

Comment:
I do understand the motivation for this change, but an old form was chosen for performance. This method is called so many times during query execution that micro-optimizations which are usually unnecessary do make sense here. Can we optionally ignore code style checks for such intentional things?

Reply:
This actually is not CS related directly. But as I required declared strict types after `<?php` this part stopped working as it's not type-safe. I'll look into it again and see if I come up with more performant way.

Reply:
Two notes on this:

1. Such micro-optimization deserve a line comment.
2. Yes, there are comments to disable style-checks on a per-line basis. Depends on the tool that is used.